test/tools/jsontests/LegacyTestsBoost.cpp

```diff
@@ -0,0 +1,221 @@
+#include <test/tools/jsontests/BlockChainTests.h>
+#include <test/tools/jsontests/StateTests.h>
+#include <test/tools/libtesteth/TestHelper.h>
+
+#include <boost/test/unit_test.hpp>
+
+using namespace dev::test;
+namespace fs = boost::filesystem;
+
+fs::path StateTestSuiteLegacyConstantinople::suiteFolder() const
+{
+    return "LegacyTests/Constantinople/GeneralStateTests";
+}
+
+fs::path StateTestSuiteLegacyConstantinople::suiteFillerFolder() const
+{
+    return "LegacyTests/Constantinople/GeneralStateTestsFiller";
+}
+
+fs::path BCGeneralStateTestsSuiteLegacyConstantinople::suiteFolder() const
+{
+    return "LegacyTests/Constantinople/BlockchainTests/GeneralStateTests";
+}
+
+fs::path BCGeneralStateTestsSuiteLegacyConstantinople::suiteFillerFolder() const
+{
+    return "LegacyTests/Constantinople/BlockchainTestsFiller/GeneralStateTests";
+}
+
+template <class T>
+class LegacyTestFixtureBase
+{
+public:
+    LegacyTestFixtureBase(bool _onlyRunIfOptionsAllSet)
+    {
+        T suite;
+        if (Options::get().fillchain || Options::get().filltests)
+            BOOST_FAIL("Legacy tests are sealed and not refillable!");
+
+        string casename = boost::unit_test::framework::current_test_case().p_name;
+        boost::filesystem::path suiteFillerPath = suite.getFullPathFiller(casename).parent_path();
+        static vector<string> const timeConsumingTestSuites{
```

Comment:
Move this into a single common place, so that it's not duplicated with regular state tests.

It's a maintenance pain to remember that it exists in several places, when there's need to modify it.

Reply:
it is not global. StateTests has their own list which might change. LegacyStateTests has their current one. Currently Legacy tests and State tests has the same. But this might change.